addons/calendar/controllers/main.py

```diff
@@ -1,49 +1,50 @@
 import json
-import openerp
-import openerp.http as http
-from openerp.http import request
-import openerp.addons.web.controllers.main as webmain
-import json
+
+import odoo
+import odoo.http as http
+from odoo import api
+from odoo.http import request
+from odoo.modules.registry import RegistryManager
 
 
-class meeting_invitation(http.Controller):
+class MeetingInvitation(http.Controller):
 
     @http.route('/calendar/meeting/accept', type='http', auth="calendar")
     def accept(self, db, token, action, id, **kwargs):
-        registry = openerp.modules.registry.RegistryManager.get(db)
-        attendee_pool = registry.get('calendar.attendee')
+        registry = RegistryManager.get(db)
         with registry.cursor() as cr:
-            attendee_id = attendee_pool.search(cr, openerp.SUPERUSER_ID, [('access_token', '=', token), ('state', '!=', 'accepted')])
-            if attendee_id:
-                attendee_pool.do_accept(cr, openerp.SUPERUSER_ID, attendee_id)
+            env = api.Environment(cr, odoo.SUPERUSER_ID, context=request.context)
+            attendee = env['calendar.attendee'].search([('access_token', '=', token), ('state', '!=', 'accepted')])
+            if attendee:
+                attendee.do_accept()
         return self.view(db, token, action, id, view='form')
 
     @http.route('/calendar/meeting/decline', type='http', auth="calendar")
     def declined(self, db, token, action, id):
-        registry = openerp.modules.registry.RegistryManager.get(db)
-        attendee_pool = registry.get('calendar.attendee')
+        registry = RegistryManager.get(db)
         with registry.cursor() as cr:
-            attendee_id = attendee_pool.search(cr, openerp.SUPERUSER_ID, [('access_token', '=', token), ('state', '!=', 'declined')])
-            if attendee_id:
-                attendee_pool.do_decline(cr, openerp.SUPERUSER_ID, attendee_id)
+            env = api.Environment(cr, odoo.SUPERUSER_ID, context=request.context)
+            attendee = env['calendar.attendee'].search([('access_token', '=', token), ('state', '!=', 'declined')])
+            if attendee:
+                attendee.do_decline()
         return self.view(db, token, action, id, view='form')
 
     @http.route('/calendar/meeting/view', type='http', auth="calendar")
     def view(self, db, token, action, id, view='calendar'):
-        registry = openerp.modules.registry.RegistryManager.get(db)
-        meeting_pool = registry.get('calendar.event')
-        attendee_pool = registry.get('calendar.attendee')
-        partner_pool = registry.get('res.partner')
+        registry = RegistryManager.get(db)
         with registry.cursor() as cr:
-            attendee = attendee_pool.search_read(cr, openerp.SUPERUSER_ID, [('access_token', '=', token)], [])
+            env = api.Environment(cr, odoo.SUPERUSER_ID, context=request.context)
+            attendee = env['calendar.attendee'].search_read([('access_token', '=', token)], [])
```

Comment:
search with limit=1 make sense here and then you can use attendee.partner_id... at line 41, 42